mathics/builtin/patterns.py

```diff
@@ -1332,3 +1334,25 @@ def _match(leaf):
             return (leaf.has_form(('Rule', 'RuleDelayed'), 2) or
                     leaf.has_form('List', None))
         return [leaf for leaf in leaves if _match(leaf)]
+
+
+class _StopGeneratorBaseExpressionIsFree(StopGenerator):
+    pass
+
+
+def is_free(expr, form, evaluation):
+    # for vars, rest in form.match(self, {}, evaluation, fully=False):
+    def yield_match(vars, rest):
+        raise _StopGeneratorBaseExpressionIsFree(False)
+        # return False
+
+    try:
+        form.match(yield_match, expr, {}, evaluation, fully=False)
+    except _StopGeneratorBaseExpressionIsFree as exc:
+        return exc.value
+
+    if expr.is_atom():
+        return True
+    else:
+        return expr.head.is_free(form, evaluation) and all(
+            leaf.is_free(form, evaluation) for leaf in expr.leaves)
```

Comment:
I forgot to hook this up. It's done now. I needed to move this function into `patterns.py` as the declaration of `_StopGeneratorBaseExpressionIsFree` needs `StopGenerator` which cannot be imported into `core.expression.py` as `patterns.py` depends on `core.expression.py`.